### DISCUSSION TITLE
Difference in Writing Providers in Next.js 12 and Next.js 14

### DISCUSSION THREAD

**Discussion by kemaltf** [vercel/next.js#59442] [Help]
### Summary

While I was learning Next.js 14, I noticed a difference in the way providers are written compared to Next.js 12, specifically when using the toast library to display alerts.

Next.js 12:

```
const Layout = ({ children }) => {
  return (
    <ToasterProvider>
      {/* Other components in the layout */}
      {children}
    </ToasterProvider>
  );
};
```

Next.js 14:

```
<body className={inter.className}>
  <ToasterProvider />
  {children}
</body>
```

The notable difference lies in the use of self-tagging, where children and other components are not explicitly wrapped inside the provider. 

Could someone provide more details about this? Thank you.

**Comment by icyJoseph** [comment]
Why did you do that change? Where's ToasterProvider coming from? In general, the way you have written it in Next14, which is NOT a necessary change to use Next14, the children won't have access to the Provider, and might only be able to use the Context's default value, as they regularly do in React.
